apps/src/signUpFlow/FinishTeacherAccount.tsx

```diff
@@ -65,62 +97,86 @@ const FinishTeacherAccount: React.FunctionComponent<{
+          <BodyTwoText>{locale.tailor_experience()}</BodyTwoText>
+        </div>
+        <fieldset className={style.inputContainer}>
+          <div>
+            <TextField
+              name="displayName"
+              label={locale.what_do_you_want_to_be_called()}
+              className={style.nameInput}
+              value={name}
+              placeholder={locale.msCoder()}
+              onChange={onNameChange}
+            />
+            <BodyThreeText className={style.displayNameSubtext}>
+              {locale.this_is_what_your_students_will_see()}
             </BodyThreeText>
+            {showNameError && (
+              <BodyThreeText className={style.errorMessage}>
+                {locale.display_name_error_message()}
+              </BodyThreeText>
+            )}
+          </div>
+          <SchoolDataInputs usIp={usIp} includeHeaders={false} />
+          {showGDPR && (
+            <div>
+              <BodyThreeText className={style.teacherKeepMeUpdated}>
+                <strong>{locale.data_transfer_notice()}</strong>
+              </BodyThreeText>
+              <Checkbox
+                name="gdprAcknowledge"
+                label={locale.data_transfer_agreement_teacher()}
+                checked={gdprChecked}
+                onChange={onGDPRChange}
+                size="s"
+              />
+            </div>
           )}
-        </div>
-        <SchoolDataInputs usIp={usIp} includeHeaders={false} />
-        <div>
-          <BodyThreeText className={style.teacherKeepMeUpdated}>
-            <strong>{locale.keep_me_updated()}</strong>
-          </BodyThreeText>
-          <Checkbox
-            name="userEmailOptIn"
-            label={locale.get_informational_emails()}
-            checked={emailOptInChecked}
-            onChange={onEmailOptInChange}
+          <div>
+            <BodyThreeText className={style.teacherKeepMeUpdated}>
+              <strong>{locale.keep_me_updated()}</strong>
+            </BodyThreeText>
+            <Checkbox
+              name="userEmailOptIn"
+              label={locale.get_informational_emails()}
+              checked={emailOptInChecked}
+              onChange={onEmailOptInChange}
+              size="s"
+            />
+            <BodyFourText className={style.emailOptInFootnote}>
+              <strong>{locale.note()}</strong>{' '}
+              {locale.after_creating_your_account()}
+            </BodyFourText>
+          </div>
+        </fieldset>
+        <div className={style.finishSignUpButtonContainer}>
+          <Button
+            className={style.finishSignUpButton}
+            color={buttonColors.purple}
+            type="primary"
+            onClick={() => sendFinishEvent()}
+            text={locale.go_to_my_account()}
+            iconRight={{
+              iconName: 'arrow-right',
+              iconStyle: 'solid',
+              title: 'arrow-right',
+            }}
+            disabled={name === '' || !gdprValid}
```

Comment:
This line is changed to include a new check for gdpr: rest is unchanged

Reply:
I wonder if this button should also be disabled while querying for gdpr?

Reply:
I added a new state param called isGdprLoaded and added it to the gdprValid memo!